frontend/components/site/siteSettings.js

```diff
@@ -116,7 +116,8 @@ class SiteSettings extends React.Component {
                 <input
                   name="domain"
                   className="form-control"
-                  type="text"
+                  type="url"
+                  pattern="https?://.+"
```

Comment:
Using `type="url"` and `pattern="https?://.+"` are primarily so that feedback is provided to a user if they enter an invalid URL. They do not mitigate the XSS issue on their own because an attacker could still set "bad" URLs directly via the API.